File: backend/schemas/slick.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from schemas.common import GeoJSONMultiPolygon, GeoJSONPoint


class OilSlick(BaseModel):
    model_config = ConfigDict(protected_namespaces=())

    id: UUID
    event_id: Optional[str] = None
    scene_id: UUID
    acquisition_timestamp: Optional[datetime] = None
    processing_timestamp: Optional[datetime] = None
    source: Optional[str] = None
    geometry: GeoJSONMultiPolygon
    crs: Optional[str] = None
    bbox: Optional[list[float]] = None
    area_km2: float
    perimeter_km: float
    centroid: GeoJSONPoint
    orientation_deg: Optional[float]
    confidence: float = Field(ge=0, le=1)
    possible_lookalike: bool
    lookalike_reason: Optional[str]
    model_version: str
    v4_threshold: Optional[float] = None
    v3_threshold: Optional[float] = None
    candidate_count: Optional[int] = None
    accepted_candidates: Optional[int] = None
    source_width: Optional[int] = None
    source_height: Optional[int] = None
    created_at: datetime

File: frontend/src/types/slick.ts
import type { GeoJSONMultiPolygon, GeoJSONPoint } from "./geo";

export interface OilSlick {
  id: string;
  event_id?: string;
  scene_id: string;
  acquisition_timestamp?: string;
  processing_timestamp?: string;
  source?: string;
  geometry: GeoJSONMultiPolygon;
  crs?: string;
  bbox?: number[];
  area_km2: number;
  perimeter_km: number;
  centroid: GeoJSONPoint;
  orientation_deg?: number;
  confidence: number;
  possible_lookalike: boolean;
  lookalike_reason?: string;
  model_version: string;
  v4_threshold?: number;
  v3_threshold?: number;
  candidate_count?: number;
  accepted_candidates?: number;
  source_width?: number;
  source_height?: number;
  created_at?: string;
}
